modules/ballerina-core/src/main/java/org/ballerinalang/util/semantics/SemanticAnalyzer.java

```diff
@@ -456,6 +456,21 @@ public void visit(BallerinaFunction function) {
 
             BlockStmt blockStmt = function.getCallableUnitBody();
             blockStmt.accept(this);
+
+            if (function.getReturnParameters().length > 0) {
+                Statement[] stmts = blockStmt.getStatements();
+                for (int i = 0; i < stmts.length; i++) {
```

Comment:
Can you please explain the reason for having this for loop here?  AFAIU, you execute the same logic in the visit Block statement method